### PR TITLE
[openimageio] fix linux cross-compile

### PR DESCRIPTION
**Pull request by russelltg**
- [openimageio] fix linux cross compile
- ./vcpkg x-add-version --all
openimageio detects if it can do atomic without linking to -latomic ops with a try_run, which is not supported when cross-compiling. This is always true on linux, so set it to allow for crosscompiling

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/openimageio/portfile.cmake [hunk 1]
```diff
@@ -48,10 +48,15 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
         tools       USE_QT
 )
 
+if (VCPKG_TARGET_IS_LINUX)
+    set(PLATFORM_OPTIONS -DCOMPILER_SUPPORTS_ATOMIC_WITHOUT_LIBATOMIC_EXITCODE=0)
+endif()
```
Comment (dg0yt) [inline ports/openimageio/portfile.cmake:53]: I'm under the impression that we shouldn't add more assumptions for particular targets, but rather interact with upstream if *running* an artifact is necessary at all. AFAIU all they need to know is whether `-latomic` is needed at link time.

Comment (FrankXie05) [reply ports/openimageio/portfile.cmake:53]: @russelltg We need to know if this is supported behavior upstream.

**Comment by russelltg** [comment]
I agree that that is the proper solution. Opened ticket upstream https://github.com/OpenImageIO/oiio/issues/3773

**Comment by russelltg** [comment]
Going to close this as I've upstreamed a better fix, so will wait for a release then I'll file a PR with a port update

**Pull request closed without merging**